### PR TITLE
Task 03: get api/categories

### PR DESCRIPTION
**Pull request by omcmanus1**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+const db = require("./db/connection");
+const express = require("express");
+const app = express();
+
+const { errorHandler500 } = require("./controllers/error-handling-controllers");
+const { getCategories } = require("./controllers/categories-controllers");
+
+app.use(express.json());
```
Comment (emilyb93) [inline app.js]: remove this until you need it

File: app.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+app.use((req, res, next) => next());
```
Comment (emilyb93) [inline app.js]: This is not doing anything, its just moving straight to your 500 handler, without the error, just remove it

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    return request(app).get("/api/categories").expect(200);
+  });
+  test("GET: should respond with an array", () => {
+    return request(app)
+      .get("/api/categories")
+      .expect(200)
+      .then(({ body }) => {
+        const { categories } = body;
+        expect(categories).toBeInstanceOf(Array);
+      });
```
Comment (emilyb93) [inline __tests__/app.test.js]: These are redundant tests as you are expecting 200 in the other tests, and you are using a forEach which is an array method, which proves this for us

File: models/categories-models.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const db = require("../db/connection");
+
+exports.selectCategories = () => {
+  let queryString = `SELECT * FROM categories`;
```
Comment (emilyb93) [inline models/categories-models.js]: This should be a const since you are not reassigning the value

**Review by emilyb93 — CHANGES_REQUESTED** [review]
see comments

File: __tests__/app.test.js (hunk 3 above)
Comment (omcmanus1) [reply __tests__/app.test.js]: I have refactored this, but not sure if correct - assertions are now all inside the .then but unable to nest new tests within this. Am I on the wrong track here?

File: __tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        //       const { categories } = body;
+        //       categories.forEach((category) => {
+        //         expect(category).toHaveProperty("slug");
+        //         expect(category).toHaveProperty("description");
+        //       });
+        //     });
+        // });
+      });
+  });
+  test("GET: should respond with a 400 error if incorrect endpoint is specified", () => {
```
Comment (yalicee) [inline __tests__/app.test.js]: description says 400 but expect is 404

**Review by yalicee — APPROVED** [review]
Looks good so I am happy to approve :) 
Please delete all of your commented out code though.

**Pull request merged**